### PR TITLE
games-roguelike/dwarf-fortress: merge two PR's

### PR DESCRIPTION
**Pull request by ezzieyguywuf**
This PR is intended to merge the work done in #20242 and #20534.

- games-roguelike/dwarf-fortress: version bump to 0.47.05, bugfix
- games-roguelike/dwarf-fortress: merge changes from PR#20534

Let's see if this works:

Closes: #20242
Closes: #20534

### CODE REVIEW

File: games-roguelike/dwarf-fortress/dwarf-fortress-0.47.05.ebuild [hunk 1]
```diff
@@ -73,12 +89,11 @@ src_install() {
 pkg_postinst() {
 	elog "System-wide Dwarf Fortress has been installed to ${gamesdir}. This is"
 	elog "symlinked to ~/.dwarf-fortress when dwarf-fortress is run."
-	elog "For more information on what exactly is replaced, see /usr/bin/dwarf-fortress."
-	elog "Note: This means that the primary entry point is /usr/bin/dwarf-fortress."
+	elog "For more information on what exactly is replaced, see ${EPREFIX}/usr/bin/dwarf-fortress."
+	elog "Note: This means that the primary entry point is ${EPREFIX}/usr/bin/dwarf-fortress."
 	elog "Do not run ${gamesdir}/libs/Dwarf_Fortress."
```
Comment (ionenwks) [inline games-roguelike/dwarf-fortress/dwarf-fortress-0.47.05.ebuild:94]: Is it necessary to show this wall of elog every time? Is once using REPLACING_VERSIONS enough? Or maybe go with [readme.gentoo-r1](https://devmanual.gentoo.org/eclass-reference/readme.gentoo-r1.eclass/index.html)?

File: games-roguelike/dwarf-fortress/dwarf-fortress-0.47.05.ebuild [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
-EAPI=6
+EAPI=7
 
-inherit versionator toolchain-funcs
+inherit flag-o-matic optfeature prefix toolchain-funcs
 
-MY_PV=$(replace_all_version_separators _ "$(get_version_component_range 2-)")
-MY_PN=df
+MY_PV=$(ver_cut 2- $(ver_rs 2- '_'))
+MY_PN="df"
```
Comment (ionenwks) [inline games-roguelike/dwarf-fortress/dwarf-fortress-0.47.05.ebuild:9]: Drop MY_PN? df is shorter than the variable name and used only twice.

**Review by ionenwks — COMMENTED** [review]
Thanks for the PR, looks good.. figure I might as well review & merge it myself now that I can :) No rush, so look at it when you you want to escape from rstudio.

Just a *few* nitpicks.

When done would be nice to merge the two commits using `Co-authored-by:`, and not delete the previous version.

**Comment by orlitzky** [comment]
Ah, I'm glad there's a good fix here. I found that the segfault and infinite loop only happen with `-O1` or higher, and was going to suggest a `replace-flags` workaround. But the missing return values are also obviously wrong so that wasn't a very satisfying solution.

**Comment by thesamesam** [comment]
Any interest in this still? Would be great to have a newer DF finally. Most of the comments are quite small.

**Comment by ezzieyguywuf** [comment]
Yes definitely interested in closing this up.

I actually just stumbled across these comments the other day and mentioned
to ionen that I'd be working on them 😀

**Pull request closed without merging**